modules/core/src/main/java/org/apache/ignite/internal/processors/metric/export/MetricResponse.java

```diff
@@ -0,0 +1,484 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.metric.export;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.BiConsumer;
+
+import org.apache.ignite.plugin.extensions.communication.Message;
+import org.apache.ignite.plugin.extensions.communication.MessageReader;
+import org.apache.ignite.plugin.extensions.communication.MessageWriter;
+import org.jetbrains.annotations.Nullable;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.ignite.internal.processors.metric.export.MetricType.BOOLEAN;
+import static org.apache.ignite.internal.processors.metric.export.MetricType.DOUBLE;
+import static org.apache.ignite.internal.processors.metric.export.MetricType.HISTOGRAM;
+import static org.apache.ignite.internal.processors.metric.export.MetricType.HIT_RATE;
+import static org.apache.ignite.internal.processors.metric.export.MetricType.INT;
+import static org.apache.ignite.internal.processors.metric.export.MetricType.LONG;
+import static org.apache.ignite.internal.util.GridUnsafe.BYTE_ARR_OFF;
+import static org.apache.ignite.internal.util.GridUnsafe.copyMemory;
+import static org.apache.ignite.internal.util.GridUnsafe.getInt;
+import static org.apache.ignite.internal.util.GridUnsafe.getLong;
+import static org.apache.ignite.internal.util.GridUnsafe.getShort;
+import static org.apache.ignite.internal.util.GridUnsafe.putInt;
+import static org.apache.ignite.internal.util.GridUnsafe.putLong;
+import static org.apache.ignite.internal.util.GridUnsafe.putShort;
+
+/**
+ * Compact data structure.
+ *
+ * Message format: header | [schema] | [data]
+ *
+ * Header:
+ *  0 - int - message size in bytes.
+ *  4 - short - version of protocol.
+ *  6 - long - timestamp.
+ * 14 - int - version of schema.
+ * 18 - int - offset of schema frame (0xFFFFFFFF if no schema frame in message).
+ * 22 - int - size of schema frame in bytes (0 if no schema frame in message).
+ * 26 - int - offset of data frame (0xFFFFFFFF if no data farame in message).
+ * 30 - int - size of data frame in bytes (0 if no data in message).
+ * 34 - UUID - cluster ID (two long values: most significant bits, then least significant bits).
+ * 50 - int - size of user tag in bytes (0 if user tag isn't defined).
+ * 54 - byte[] - user tag.
+ * 54 + user tag size - int - size of consistent ID in bytes.
+ * 54 + user tag size + 4 - byte[] - consistent ID ()
+ *
+ * Schema:
+ * 0 - byte - particular metric type.
+ * 1 - int - size of particular metric name in bytes.
+ * 5 - bytes - particular metric name
+ *
+ * Data:
+ * Accordingly to schema.
+ *
+ */
+public class MetricResponse implements Message {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** Protocol version 1. */
+    private static final short PROTO_VER_1 = 1;
```

Comment:
why we have private static const with protocol version in several places?

Reply:
Removed this one, added a reference to request.